common/perf-test-core/src/main/java/com/azure/perf/test/core/MockEventProcessor.java

```diff
@@ -0,0 +1,150 @@
+    private final AtomicReference<ScheduledExecutorService> scheduler = new AtomicReference<>();
+
+    /**
+     * Creates an instance of a mock event processor
+     *
+     * @param partitions the number of partitions
+     * @param maxEventsPerSecond the maximum events per second to send, optional.
+     * @param errorAfter the duration after which processor should error out, optional.
+     * @param processError the consumer to process the error.
+     * @param processEvent the consumer to process the event.
+     */
+    public MockEventProcessor(int partitions, int maxEventsPerSecond, Duration errorAfter,
+                              Consumer<MockErrorContext> processError, Consumer<MockEventContext> processEvent) {
+        this.processError = processError;
+        this.processEvent = processEvent;
+        this.partitions = partitions;
+        this.maxEventsPerSecond = maxEventsPerSecond;
+        this.maxEventsPerSecondPerPartition = ((double) maxEventsPerSecond) / partitions;
+        this.errorAfter = errorAfter;
+        this.errorLock = new ReentrantLock();
+
+        mockEventContexts = new MockEventContext[partitions];
+        IntStream.range(0, partitions).boxed().forEach(integer -> {
+            mockEventContexts[integer] = new MockEventContext(integer, "Hello");
+        });
+        this.eventsRaised = new int[partitions];
+    }
+
+    /**
+     * Starts the event processor.
+     */
+    public synchronized void start() {
+        eventsRaised = new int[eventsRaised.length];
+        process = true;
+        errorRaised = false;
+        startTime = System.nanoTime();
+        ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+        scheduler.set(executor);
+
+        Double jitterInMillis =
+            ThreadLocalRandom.current().nextDouble() * TimeUnit.SECONDS.toMillis(0);
+
+        runner.set(scheduler.get().schedule(this::processEvents,
+            jitterInMillis.longValue(), TimeUnit.MILLISECONDS));
+    }
+
+    private Mono<Void> processEvents() {
+        while (process) {
+            for (int i = 0; i < partitions; i++) {
+                process(i);
+            }
+        }
+        return Mono.empty();
+    }
+
+    private void process(int partition) {
+        MockEventContext mockEventContext = mockEventContexts[partition];
+
+        if (maxEventsPerSecond > 0) {
+            while (process) {
+                long elapsedTime = (System.nanoTime() - startTime);
+                if (errorAfter != null && !errorRaised
+                    && (errorAfter.compareTo(Duration.ofNanos(elapsedTime)) < 0)) {
+                    errorLock.lock();
+                    if (!errorRaised) {
+                        processError(partition, new IllegalStateException("Test Exception"));
+                        errorRaised = true;
+                    }
+                }
+                int eventsSent = eventsRaised[partition];
```

Comment:
This should be in an `else` block of `if (errorAfter` above.  See the .NET version:

https://github.com/Azure/azure-sdk-for-net/blob/main/common/Perf/Azure.Sample.Perf/Event/MockEventProcessor.cs#L99